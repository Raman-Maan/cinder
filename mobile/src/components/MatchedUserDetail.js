import React, { Component } from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { Container, Content, Card, CardItem, Text } from 'native-base';

import cinder from '../theme/variables/cinder';

const styles = StyleSheet.create({
  header: {
    alignItems: 'center',
  },
  container__view: {
    flex: 1,
    alignItems: 'center',
  },
  user__about: {
    paddingLeft: '5%',
    paddingRight: '5%',
    paddingBottom: '2%',
  },
  user__image: {
    height: 300,
    width: 300,
    flex: 1,
    borderRadius: cinder.cardBorderRadius,
    backgroundColor: 'white',
    resizeMode: 'contain',
  },
  user__age: {
    fontStyle: 'italic',
    color: 'grey',
  },
  user__matched_date: {
    color: 'grey',
  },
  card: {
    alignItems: 'center',
    marginBottom: '2%',
  },
});

export default class UserDetails extends Component {
  static navigationOptions({ navigation }) {
    return {
      title: navigation.state.params.userName,
    };
  }

  constructor(props) {
    super(props);

    const {
      userName, userBio, primaryPic, matchDate, userAge,
    } = this.props.navigation.state.params;

    this.state = {
      name: userName,
      bio: userBio,
      img: primaryPic,
      matchDate: new Date(matchDate).toISOString().split('T')[0],
      age: userAge,
    };
  }

  render() {
    return (
      <Container>
        <View style={styles.container__view}>
          <Content>
            <Card style={styles.card}>
              <CardItem header style={styles.header}>
                <Text style={styles.user__matched_date}> Matched on {this.state.matchDate}</Text>
              </CardItem>
              <CardItem>
                <Image style={styles.user__image} source={{ uri: this.state.img }} />
              </CardItem>
              <CardItem>
                <Text style={styles.user__age}>{this.state.name}, {this.state.age}</Text>
              </CardItem>
              <Text style={styles.user__about}>{this.state.bio}</Text>
            </Card>
          </Content>
        </View>
      </Container >
    );
  }
}
