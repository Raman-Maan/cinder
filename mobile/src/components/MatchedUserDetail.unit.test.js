import React from 'react';
import renderer from 'react-test-renderer';

import MatchedUserDetail from './MatchedUserDetail';

describe('<MatchedUserDetail />', () => {
  const testData =
    {
      userID: 1,
      userName: 'Mac Miller',
      age: 21,
      primaryPic: 'https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b',
      userBio: 'He raps, he does some other stuff too but mostly just that',
      matchDate: '2018-02-03T18:09:05.000Z',
    };

  it('should render when empty', () => {
    const rendered = renderer.create(<MatchedUserDetail data={[]} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders with data', () => {
    const rendered = renderer.create(<MatchedUserDetail data={testData} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
