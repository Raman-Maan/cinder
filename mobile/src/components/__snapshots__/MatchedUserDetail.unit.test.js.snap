// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchedUserDetail /> renders with data 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={undefined}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      viewIsInsideTabBar={false}
      virtual={undefined}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flex": 1,
                "flexWrap": "nowrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "alignItems": "center",
                "marginBottom": "2%",
              },
            ]
          }
          virtual={undefined}
        >
          <View
            header={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": null,
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 15,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                },
                Object {
                  "alignItems": "center",
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "500",
                  },
                  Object {
                    "color": "grey",
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
               Matched on 
              2018-02-03
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Image
              source={
                Object {
                  "uri": "https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b",
                }
              }
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 20,
                  "flex": 1,
                  "height": 300,
                  "resizeMode": "contain",
                  "width": 300,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "grey",
                    "fontStyle": "italic",
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              Mac Miller
              , 
              21
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                },
                Object {
                  "paddingBottom": "2%",
                  "paddingLeft": "5%",
                  "paddingRight": "5%",
                },
              ]
            }
            uppercase={false}
            virtual={undefined}
          >
            He raps, he does some other stuff too but mostly just that
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
